packages/neon_framework/lib/src/widgets/custom_background.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'package:neon_framework/blocs.dart';
+import 'package:neon_framework/src/utils/global_options.dart';
+import 'package:neon_framework/src/utils/hex_color.dart';
+import 'package:neon_framework/src/utils/provider.dart';
+import 'package:neon_framework/src/widgets/image.dart';
+import 'package:neon_framework/src/widgets/options_collection_builder.dart';
+import 'package:nextcloud/core.dart' as core;
+
+/// Displays the custom background if the user has it enabled.
+///
+/// The background will be loaded from the server.
+/// It might be a single color or a full image.
+class NeonCustomBackground extends StatelessWidget {
+  const NeonCustomBackground({
+    required this.child,
+    super.key,
+  });
+
+  /// Widget displayed on top of the custom background.
+  final Widget child;
+
+  @override
+  Widget build(final BuildContext context) {
+    final globalOptions = NeonProvider.of<GlobalOptions>(context);
+    final accountsBloc = NeonProvider.of<AccountsBloc>(context);
+
+    return OptionsCollectionBuilder(
+      valueListenable: globalOptions,
+      builder: (final context, final options, final _) {
+        if (!options.themeUseNextcloudTheme.value || !options.themeCustomBackground.value) {
+          return child;
+        }
+
+        return StreamBuilder(
+          stream: accountsBloc.activeAccount,
+          builder: (final context, final activeAccountSnapshot) =>
+              ResultBuilder<core.OcsGetCapabilitiesResponseApplicationJson_Ocs_Data?>.behaviorSubject(
+            subject: activeAccountSnapshot.hasData
+                ? accountsBloc.getCapabilitiesBlocFor(activeAccountSnapshot.data!).capabilities
+                : null,
+            builder: (final context, final capabilitiesSnapshot) {
+              final theme = capabilitiesSnapshot.data?.capabilities.themingPublicCapabilities?.theming;
```

Comment:
The server theme is already queried on startup.
I think we should insert it as a theme extension in the [AppTheme](https://github.com/nextcloud/neon/blob/119dfe912c16fd0cd82dcafb790d9547c8efd258/packages/neon_framework/lib/src/theme/theme.dart).